Libplanet/Assets/FungibleAssetValue.cs

```diff
@@ -0,0 +1,530 @@
+        /// by <paramref name="divisor"/>.</returns>
+        /// <exception cref="ArgumentException">Thrown when two values do not have the same
+        /// <see cref="Currency"/>.</exception>
+        /// <exception cref="DivideByZeroException">Thrown when the <paramref name="divisor"/> is
+        /// zero.</exception>
+        [Pure]
+        public static FungibleAssetValue operator %(
+            FungibleAssetValue dividend,
+            FungibleAssetValue divisor)
+        {
+            if (!dividend.Currency.Equals(divisor.Currency))
+            {
+                throw new ArgumentException(
+                    "Cannot be divided by a heterogeneous currency: " +
+                    $"{dividend.Currency} \u2260 {divisor.Currency}."
+                );
+            }
+
+            return new FungibleAssetValue(dividend.Currency, dividend.RawValue % divisor.RawValue);
+        }
+
+        /// <summary>
+        /// Divides the value by <paramref name="divisor"/>, returns the quotient, and returns
+        /// the <paramref name="remainder"/> in an output parameter.
+        /// </summary>
+        /// <param name="divisor">The number to divide by.</param>
+        /// <param name="remainder">When this method returns (without any exception), the remainder
+        /// after dividing the value by <paramref name="divisor"/>.  This parameter is passed
+        /// uninitialized.</param>
+        /// <returns>The quotient of the division.</returns>
+        /// <exception cref="DivideByZeroException">Thrown when the <paramref name="divisor"/> is
+        /// <c>0</c> (zero).</exception>
+        [Pure]
+        public FungibleAssetValue DivRem(BigInteger divisor, out FungibleAssetValue remainder)
+        {
+            BigInteger q = BigInteger.DivRem(RawValue, divisor, out BigInteger rem);
+            remainder = new FungibleAssetValue(Currency, rem);
+            return new FungibleAssetValue(Currency, q);
+        }
+
+        /// <summary>
+        /// Divides the value by <paramref name="divisor"/>, returns the quotient, and returns
+        /// the <paramref name="remainder"/> in an output parameter.
+        /// </summary>
+        /// <param name="divisor">The value to divide by.</param>
+        /// <param name="remainder">When this method returns (without any exception), the remainder
+        /// after dividing the value by <paramref name="divisor"/>.  This parameter is passed
+        /// uninitialized.</param>
+        /// <returns>The quotient of the division.</returns>
+        /// <exception cref="ArgumentException">Thrown when two values do not have the same
+        /// <see cref="Currency"/>.</exception>
+        /// <exception cref="DivideByZeroException">Thrown when the <paramref name="divisor"/> is
+        /// zero.</exception>
+        [Pure]
+        public BigInteger DivRem(FungibleAssetValue divisor, out FungibleAssetValue remainder)
```

Comment:
Is this operation(`someFAV.DivRem(someOtherFAV, out FAV rem)`) really needed?

Reply:
It's intended to be parallel with `BigInteger`… a overloaded method which returns a remainder through a tuple instead `out` parameter does not exist in `BigInteger` but I added here for convenience' sake.

Reply:
I was wondering there are situations in which one FAV needs to be divided by another FAV (not a number). I guess we can cover most of cases by `.DivRem(BigInteger, out FungibleAssetValue)`.